pkg/oauth2/token_source.go

```diff
@@ -7,9 +7,23 @@ import (
 	"sync"
 	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
 	"golang.org/x/oauth2"
 )
 
+var (
+	grantsTotal = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "password_credentials_grants_total",
+			Help: "Tracks the number of resource owner password credential grants.",
+		},
+	)
+)
+
+func init() {
+	prometheus.MustRegister(grantsTotal)
```

Comment:
Could we not use the global registry? I'd like us to generally move away from this pattern, and we should be a role model for the rest of the teams.

Reply:
good idea. we can do a head-start here :+1: I suggest to dependency-inject the registry here, then gradually factor all the other places, wdyt?

Reply:
that works for me

Reply:
addressed, ptal